packages/data-context/src/sources/FileDataSource.ts

```diff
@@ -10,24 +10,22 @@ const debug = Debug('cypress:data-context:sources:FileDataSource')
 export class FileDataSource {
   constructor (private ctx: DataContext) {}
 
-  readFile (absoluteFilePath: string) {
-    return this.fileLoader.load(absoluteFilePath).catch((e) => {
-      this.fileLoader.clear(absoluteFilePath)
-      throw e
-    })
-  }
+  async checkIfFileExists (relativePath: string) {
+    assert(this.ctx.currentProject, `Cannot checkIfFileExists without active project`)
 
-  readFileInProject (relativeFilePath: string) {
-    assert(this.ctx.currentProject, 'Cannot readFileInProject without currentProject')
+    const filePath = path.join(this.ctx.currentProject, relativePath)
 
-    return this.readFile(path.join(this.ctx.currentProject, relativeFilePath))
+    try {
+      return await this.ctx.fs.stat(filePath)
+    } catch {
+      return null
+    }
   }
 
-  readJsonFile<Result = unknown> (absoluteFilePath: string) {
-    return this.jsonFileLoader.load(absoluteFilePath).catch((e) => {
-      this.jsonFileLoader.clear(e)
-      throw e
-    }) as Promise<Result>
+  async readFileInProject (relative: string) {
```

Comment:
`async` function that then calls `readFileSync`?

Reply:
Updated